File: apps/1d/euler/shock_entropy/plotq1.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
%%   Information you have available to you:
%%
%%     Basic information:
%%              mx:  number of points
%%    [xlow,xhigh]:  min/max values of grid
%%            meqn:  number of equations
%%            maux:  number of aux components
%%           meth1:  spatial order of accuracy
%%
%%   Grid information:
%%              xc: grid points (cell centers), size = (mx,my)
%%
%%   Solution information:
%%           qsoln:  solution sampled on mesh, size = (mx,meqn)
%%             aux:  aux components sampled on mesh, size = (mx,maux)
%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% gas constant
%% ----
%% HACK
%% ----
gamma = 1.4;
%% ----

% -------------------------------------------- %
% Reference solution (at the final time):
% Uncomment if not desired.  The data for this 
% file was produced by HYPERPYWS.
fids = fopen('reference_6000.dat', 'r' );
if(fids == -1)
    disp(['File  ', 'reference6000.dat','  not found.']);
else
    qex = fscanf(fids,'%e',[4,inf]);
    fclose( fids );
    mx_ex = length( qex );
    xex = qex( 1, : )';  qex = qex( 2:4, : )';
    pex = (gamma-1)*(qex(:,3)-0.5*(qex(:,2).^2)./qex(:,1));
end    
% -------------------------------------------- %

% -------------------------------------------- %
% Secondary solution to compare result to
fids_extra = fopen('rk4-soln-100.dat', 'r' );
if(fids_extra == -1)
  disp(['File  ', 'rk4-soln-100.dat','  not found.']);
else
    qextra = fscanf(fids,'%e',[4,inf]);
    fclose( fids );
    mx_extra = length( qextra );
    x_extra = qextra( 1, : )';  qextra = qextra( 2:4, : )';
    pextra  = (gamma-1)*(qextra(:,3)-0.5*(qextra(:,2).^2)./qextra(:,1));
end    
% -------------------------------------------- %

figure(1);
clf;
pz=plot(xc,qsoln(:,1),'bo');
set(pz,'linewidth',1);
set(pz,'markersize',8);
hold off;
axis on; box on; grid off;
%axis([-5 5 0 5]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-5:2.5:5);
set(gca,'ytick',0:1:5);
set(gca,'fontsize',16);
%t1 = title(['Density at t = ',num2str(time),'     [FINESS]']); 
t1 = title(['Density']);
set(t1,'fontsize',16);
if(fids_extra ~= -1)
    hold on;
    plot( x_extra, qextra(:,1), 'g+' );
    hold off;
end
if(fids ~= -1)
    hold on;
    plot( xex, qex(:,1), '-r' );
    hold off;
end
l1 = legend('Lax-Wendroff', 'Runge-Kutta', 'Reference');
set(l1, 'Location', 'NorthWest');


figure(2);
clf;
press = (gamma-1).*(qsoln(:,5)-0.5*(qsoln(:,2).^2 + ...
                                    qsoln(:,3).^2 + qsoln(:, ...
                                                  4).^2)./qsoln(:,1));
pz=plot(xc,press,'bo');
set(pz,'linewidth',1);
set(pz,'markersize',8);
hold off;
axis on; box on; grid off;
%axis([-5 5 0 12]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-5:2.5:5);
set(gca,'ytick',0:2:12);
set(gca,'fontsize',16);
%t1 = title(['Pressure at t = ',num2str(time),'     [FINESS]']); 
t1 = title(['Pressure']);
set(t1,'fontsize',16);

if(fids_extra ~= -1)
    hold on;
    plot( x_extra, pextra(:,1), 'g+' );
    hold off;
end
if(fids ~= -1)
    hold on;
    plot( xex, pex, '-r' );
    hold off;
end
l2 = legend('Lax-Wendroff', 'Runge-Kutta', 'Reference');
set(l2,'Location', 'Best');


figure(3);
clf;
pz=plot(xc,qsoln(:,2)./qsoln(:,1),'bo');
set(pz,'markersize',8);
set(pz,'linewidth',1);
hold off;
axis on; box on; grid off;
%axis([-5 5 -0.5 3]);
set(gca,'plotboxaspectratio',[1.5 1 1]);
set(gca,'xtick',-5:2.5:5);
set(gca,'ytick',0:1:3);
set(gca,'fontsize',16);
%t1 = title(['u^1(x,t) at t = ',num2str(time),'     [FINESS]']); 
t1 = title(['Velocity']);
set(t1,'fontsize',16);

if(fids_extra ~= -1)
    hold on;
    plot( x_extra, qextra(:,2)./qextra(:,1), 'g+' );
    hold off;
end
if(fids ~= -1)
    hold on;
    plot( xex, qex(:,2)./qex(:,1), '-r' );
    hold off;
end
l3 = legend('Lax-Wendroff', 'Runge-Kutta', 'Reference');
set( l3, 'Location', 'South' );


% Save the pretty pictures!
print(1, '-depsc', 'shock_entropy_density.eps'  );
print(2, '-depsc', 'shock_entropy_pressure.eps' );
print(3, '-depsc', 'shock_entropy_velocity.eps' );

File: apps/1d/euler/shock_entropy/plotq1.py
from __future__ import print_function  # used for printing to file

#-----------------------------------------------------------------------------#
def plotq1(xc, qsoln, auxsoln, time, ini_params, m ):
    
    import matplotlib.pyplot as plt
    import numpy as np

    meqn     = ini_params['meqn']
    maux     = ini_params['maux']
    mx       = ini_params['mx']

    # Pull gas constant (or read it if it hasn't already been read)
    if( not ini_params.has_key('gamma') ):

        import ConfigParser
        config = ConfigParser.RawConfigParser()
        config.read( ini_params['output_dir'] + '/parameters.ini' )
        ini_params['gamma'] = config.getfloat('euler', 'gamma' )

    gamma   = ini_params['gamma']
    gp1     = gamma+1.
    gm1     = gamma-1.

    # DENSITY
    m=0
    plt.figure(1)
    plt.clf()
    plt.gca().set_aspect('auto')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([0.39, 5.1])
    plt.plot(xc,qsoln[:,m],'bo')
    tmp1 = "".join(("Density at t = ",str(time)))
    title = "".join((tmp1,"     [FINESS]"))
    plt.title(title)
    plt.draw()

    # PRESSURE
    m=1

    press = (gamma-1)*(qsoln[:,4]-0.5*(
        qsoln[:,1]**2 + qsoln[:,2]**2 + qsoln[:,3]**2)/qsoln[:,0])
    plt.figure(2)
    plt.clf()
    plt.gca().set_aspect('auto')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([-0.01, 12.10])
    plt.plot(xc,press,'bo')
    tmp1 = "".join(("Pressure at t = ",str(time)))
    title = "".join((tmp1,"     [FINESS]"))
    plt.title(title)
    plt.draw()
 
    # Velocity
    m=2
    u = np.zeros(mx,float)
    u = qsoln[:,1] / qsoln[:,0]
        
    plt.figure(3)
    plt.clf()
    plt.gca().set_aspect('auto')
    plt.gca().set_xlim([xc[0],xc[mx-1]])
    plt.gca().set_ylim([-0.11,  3.10])
    plt.plot(xc,u,'bo')
    tmp1 = "".join(("Velocity at t = ",str(time)))
    title = "".join((tmp1,"     [FINESS]"))
    plt.title(title)
    plt.draw()


    # print data to file.
    data = np.array( [xc, qsoln[:,0], qsoln[:,1], qsoln[:,4] ] )
    fmt   = '%1.15e'
    with open( 'soln.dat', 'wb' ) as f:
      #print( fmt % time, file=f )         # time instant on first row
      np.savetxt( 'soln.dat', np.transpose( data ), fmt=fmt )
 
#-----------------------------------------------------------------------------#
